src/components/shared/Modal.js

```diff
@@ -23,38 +23,50 @@ export default class Modal extends Component {
     self.onClick = this.onClick.bind(this);
   }
 
-  componentWillUnmount() {
-    const shortcuts = this.context.shortcuts;
-    shortcuts.off("Escape");
-    shortcuts.off(L10N.getStr(this.props.shortcut));
-  }
-
-  componentDidMount() {
-    const shortcuts = this.context.shortcuts;
-    shortcuts.on("Escape", this.props.handleClose);
-    shortcuts.on(L10N.getStr(this.props.shortcut), this.props.handleOpen);
-  }
-
   onClick(e: SyntheticEvent) {
     e.stopPropagation();
   }
 
   render() {
-    const { enabled } = this.props;
+    const { status } = this.props;
+
     return dom.div(
       {
-        className: classnames("modal-wrapper", { enabled }),
+        className: "modal-wrapper",
         onClick: this.props.handleClose
       },
       dom.div(
-        { className: classnames("modal", { enabled }), onClick: this.onClick },
+        { className: classnames("modal", status), onClick: this.onClick },
         this.props.children
       )
     );
   }
 }
 
-Modal.displayName = "Modal";
-Modal.contextTypes = {
+_Modal.displayName = "Modal";
+_Modal.contextTypes = {
   shortcuts: PropTypes.object
 };
+
+const Modal = createFactory(_Modal);
+
+type SlideProps = {
+  in: boolean,
+  children?: Children,
+  handleClose: () => any
+};
+
+export default function Slide({
+  in: inProp,
+  children,
+  handleClose
+}: SlideProps) {
+  return Transition({
+    in: inProp,
+    timeout: 175,
+    appear: true,
```

Comment:
> Normally a component is not transitioned if it shown when the <Transition> component mounts. If you want to transition on the first mount set appear to true, and the component will transition in as soon as the <Transition> mounts.

This prop is awesome but sadly there doesn't seem to be a `leave` that is the equivalent. I've opened an issue on react-transition-group requesting one to see what they recommend.